cbdr/cortex
# CE-728: Refactors for Consumer Migration

**Purpose**
The goal of this PR is to tackle various, necessary refactors prior to migrating the Consumer/A&R team to Beta Cortex.

**JIRA**
https://careerbuilder.atlassian.net/browse/CE-728

**Changes**
* Improvements and fixes
  * Transitions from `Virtus` to `dry-transaction` for Service/Transaction Layer
    * Implements new, re-usable operations
    * Implements `database_transact_operation` to wrap an `ActiveRecord` `transaction` around steps
    * Sets up IOC Container for Transaction layer
  * Migrates Media Widget Parser to `Plugins::Core`
  * Removes RSS logic
  * Removes various non-core logic
  * Dependency upgrades
  * Various refactors
  * Small Readme updates (more in forthcoming PR)

* Changes to developer setup/environment
  * N/A

* Architectural changes
  * See above!

* Migrations or Steps to Take on Production
  * N/A
  
* Library changes
  * N/A

* Side effects
  * N/A

**Screenshots**
* Before
N/A

* After
N/A

**Feature Server**
N/A

**How to Verify These Changes**
* Specific pages to visit
  * N/A

* Steps to take
  * Ensure system still works as expected

* Responsive considerations
  * N/A

**Relevant PRs/Dependencies**
  * https://github.com/cortex-cms/cortex-plugins-core/pull/70
  * https://github.com/cortex-cms/cortex-starter/pull/1

**Additional Information**
N/A
